server/src/test/java/com/objectcomputing/checkins/services/checkins/CheckInControllerTest.java

```diff
@@ -182,44 +198,32 @@ public void testGetFindByTargetYearAndTargetQtr() {
     @Test
     public void testGetFindByPdlId() {
 
-        if(!isDataSetupForGetTest){
-            setupTestData();
-            isDataSetupForGetTest = true;
-        }
+        setupTestData();
 
         HttpRequest requestFindByPdlId = HttpRequest.GET(String.format("/?pdlId=%s", testPdlId));
         List<CheckIn> responseFindByPdlId = client.toBlocking().retrieve(requestFindByPdlId, Argument.of(List.class, mockCheckIn.getClass()));  
         assertEquals(1, responseFindByPdlId.size());
-        assertEquals(testMemberId, responseFindByPdlId.get(0).getTeamMemberId());
+        assertEquals(testTeamMemberId, responseFindByPdlId.get(0).getTeamMemberId());
         assertEquals(testPdlId, responseFindByPdlId.get(0).getPdlId());
     }
 
     // POST - Valid Body
     @Test
     public void testPostSave() {
 
-        CheckIn testCheckIn = new CheckIn(UUID.randomUUID(), UUID.randomUUID(), testDate, "Q1", "2021");
-
-        when(mockCheckInRepository.save(testCheckIn)).thenReturn(testCheckIn);
+        final Map<String, Object> testMap = new HashMap<String, Object>() {{
+            put("teamMemberId", testTeamMemberId);
+            put("pdlId", testPdlId);
+            put("checkInDate", testDate);
+            put("targetQtr", testQuarter);
+            put("targetYear", testYear);
+        }};
```

Comment:
Instead of using these maps, I would put an @Consumes(MediaType.APPLICATION_JSON) on your controller and simply pass in a populated object. It will get marshalled to JSON and sent in the body (which is what our client application devs are going to prefer).

Reply:
@mkimberlin, I've added @Consumes(MediaType.APPLICATION_JSON) to the controllers, removed Hashmaps from test classes and have pruned assert statements.